diesel_cli/src/print_schema.rs

```diff
@@ -574,16 +595,33 @@ impl Display for TableDefinitions<'_> {
             writeln!(f, "{}", Joinable(foreign_key))?;
         }
 
-        if self.tables.len() > 1 {
-            write!(f, "\ndiesel::allow_tables_to_appear_in_same_query!(")?;
+        let table_groups = match self.allow_tables_in_same_query {
+            AllowTablesInSameQuery::FkRelatedTables => {
+                foreign_key_table_groups(&self.tables, &self.fk_constraints)
+            }
+            AllowTablesInSameQuery::AllTables => {
+                vec![self.tables.iter().map(|table| &table.name).collect()]
+            }
+        };
+        is_first = true;
+        for tables in table_groups {
+            if tables.len() < 2 {
+                continue;
+            }
+
+            if is_first {
+                writeln!(f)?;
+                is_first = false;
+            }
+            write!(f, "diesel::allow_tables_to_appear_in_same_query!(")?;
             {
                 let mut out = PadAdapter::new(f);
                 writeln!(out)?;
-                for table in &self.tables {
-                    if table.name.rust_name == table.name.sql_name {
-                        writeln!(out, "{},", table.name.sql_name)?;
+                for table in tables {
+                    if table.rust_name == table.sql_name {
+                        writeln!(out, "{},", table.sql_name)?;
                     } else {
-                        writeln!(out, "{},", table.name.rust_name)?;
+                        writeln!(out, "{},", table.rust_name)?;
```

Comment:
I'm aware this was there previously but... isn't that just always printing `rust_name`? 😅